src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesService.kt

```diff
@@ -74,9 +76,55 @@ class ActivitiesService(
       )
       log.debug("OffenderProgramProfile created with Nomis id = $offenderProgramReferenceId")
 
-      CreateOffenderProgramProfileResponse(offenderProgramReferenceId)
+      OffenderProgramProfileResponse(offenderProgramReferenceId)
     }
 
+  fun endOffenderProgramProfile(
+    courseActivityId: Long,
+    bookingId: Long,
+    dto: EndOffenderProgramProfileRequest
+  ): OffenderProgramProfileResponse =
+
+    offenderProgramProfileRepository.findByCourseActivityIdAndOffenderBookingIdAndAlloc(courseActivityId, bookingId)
+      ?.run {
+
+        dto.endReason?.also {
```

Comment:
I think this reads better as
`endReason = dto.endReason?.let{ programServiceEndReasonRepository.findByIdOrNull(ProgramServiceEndReason.pk(it))} ?: throw ....`

Reply:
thanks, much better 👍